exported STIRConfig.cmake can have problems with "too much" version info

On @categascoigne 's Ubuntu 22.04, we followed https://github.com/SyneRBI/SIRF/wiki/Building-SIRF-and-CIL-with-conda. This gave an error when configuring SIRF  where CMake complained about the line generated from
https://github.com/UCL/STIR/blob/7508babaf5e8147a3563f1642a47301af4c0834d/src/cmake/STIRConfig.cmake.in#L102
```
CMake Error at /home/cate/devel/buildConda/INSTALL/lib/cmake/STIR-6.2/STIRConfig.cmake:128 (find_package): 
  find_package called with invalid argument "1.14.4-3"
```
CMake was happy once we hand-edited the (exported) `STIRConfig.cmake` to use version 1.14.4.

We will therefore have to trim the version keyword(s).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -290,6 +290,8 @@ endforeach()
 set(STIR_REGISTRIES ${INSTALLED_STIR_REGISTRIES})
 
 # New create STIRConfig.cmake
+# First strip patch info from HDF5_VERSION, as on some systems it might be set as x.y.z-something, which causes problems for find_package.
+string(REGEX MATCH "^([0-9]+)\\.([0-9]+)" HDF5_VERSION_FOR_CONFIG "${HDF5_VERSION}")
 CONFIGURE_PACKAGE_CONFIG_FILE(
   src/cmake/STIRConfig.cmake.in
   "${CMAKE_BINARY_DIR}/STIRConfig.cmake"

--- a/src/cmake/STIRConfig.cmake.in
+++ b/src/cmake/STIRConfig.cmake.in
@@ -99,9 +99,9 @@ if (@HDF5_FOUND@)
     # We try to work-around this by first looking for both C and CXX, and then only for CXX.
     # Note that this only solves the problem if in the "client" code,
     # find_package(STIR) is called BEFORE find_package(HDF5)
-    find_package(HDF5 @HDF5_VERSION@ QUIET COMPONENTS C CXX)
+    find_package(HDF5 @HDF5_VERSION_FOR_CONFIG@ QUIET COMPONENTS C CXX)
   endif()
-  find_package(HDF5 @HDF5_VERSION@ REQUIRED COMPONENTS CXX )
+  find_package(HDF5 @HDF5_VERSION_FOR_CONFIG@ REQUIRED COMPONENTS CXX )
 
   set(STIR_BUILT_WITH_HDF5 TRUE)
 endif()

